content/en/docs/architecture/branch-protection.md

```diff
@@ -8,6 +8,11 @@ Branch protection is a repository setting that enforces certain rules when tryin
 the [GitHub documentation](https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/about-protected-branches)
 for more details.
 
+{{< alert title="Note" color="info" >}}
+Branch protection is not available for private repositories on GitHub’s free plan.
+This functionality is limited to public repositories on free plans or to private repositories on paid plans.
```

Comment:
I think that this isn't relevant to our users. We just need to tell them that they can't configure it for a private repo, and must instead use `unmanaged: true` in their config. They can't do anything about our GH plan, so they don't need to know that part ;)